lib/rspec-puppet/setup.rb

```diff
@@ -55,7 +55,11 @@ def self.get_module_name
     end
 
     def self.get_module_name_from_file(file)
-      p = Puppet::Parser::Lexer.new
+      if Puppet.version.to_f >= 4.0
+        p = Puppet::Pops::Parser::Lexer2.new
```

Comment:
@igalic It is not documented and not marked as @api ... I think that is a telling sign. OTOH, the old Lexer was also not considered API. However, the entire method is flawed since it is only using the lexer - it is completely unknown if the read content is valid (e.g. "class class Foo define define class" lexes just fine :-)  The correct way is to use the parser and get the parsed result (and look for validation errors). The definitions are found inside the parsed result. With the 3x parser evaluation was sort of intertwined with parsing, and you would have to evaluate the logic to find the definitions (it scanned for them when importing/evaluating) - and thus it was very tricky to find the definitions.

Albeit also not API, the EvaluatingParser class provides high level access to parsing, validation and evaluation. I would use it - to a) parse, b) get the Program instance (program) from the parsed result, c) get the definitions from it (program.definitions), c) iterate over those to find ResourceTypeDefinition instances, and HostClassDefinition instances, and get each definition.name d) split the name and get the first part. (Note that there are other model classes that are definitions than ResourceTypeDefinition and HostClassDefinition). See the file model_meta.rb for the schema/model of everything in the AST-model.